iroh/src/commands/provide.rs

```diff
@@ -238,3 +238,38 @@ impl FromStr for ProviderRpcPort {
         }
     }
 }
+
+#[derive(Debug, Clone)]
+struct TokenAuth {
+    token: Option<RequestToken>,
+}
+
+impl<D> RequestAuthorizationHandler<D> for TokenAuth {
+    fn authorize(
+        &self,
+        db: D,
+        token: Option<RequestToken>,
+        request: &Request,
+    ) -> BoxFuture<'static, anyhow::Result<()>> {
+        match &self.token {
+            None => return ().authorize(db, token, request),
```

Comment:
delegating to the empty handler behaviour here. IIUC this is the right way to approach building the `provider` node, such that the provider node's `CustomAuthHandler` type is always `TokenAuth`. If there's a cleaner way to do this no-op delegation, I'd love to know about it